integration/spark/src/main/scala/org/apache/carbondata/indexserver/IndexJobs.scala

```diff
@@ -69,6 +69,12 @@ class DistributedIndexJob extends AbstractIndexJob {
         if (null != splitFolderPath && !splitFolderPath.deleteFile()) {
           LOGGER.error("Problem while deleting the temp directory:"
             + splitFolderPath.getAbsolutePath)
+        } else {
+          // if the path build with getQueryId already exists,
+          // the splitFolderPath should be null, need delete
+          if (null == splitFolderPath) {
+            CarbonUtil.deleteTempFolderForIndexServer(indexFormat.getQueryId)
```

Comment:
We are already deleting the query id folders in this PR, why is deleting the "/tmp/indexservertmp" required then??

Reply:
if the /tmp/indexservertmp/gropid  has exists then the splitFolderPath will return null.
before our code do not deal with that scene。